networkit/cpp/centrality/DynApproxBetweenness.cpp

```diff
@@ -7,172 +8,316 @@
  */
 
 #include <networkit/centrality/DynApproxBetweenness.hpp>
+#include <networkit/auxiliary/Log.hpp>
+#include <networkit/auxiliary/NumericTools.hpp>
 #include <networkit/auxiliary/Random.hpp>
+#include <networkit/centrality/Centrality.hpp>
+#include <networkit/components/StronglyConnectedComponents.hpp>
 #include <networkit/distance/Diameter.hpp>
-#include <networkit/graph/GraphTools.hpp>
-#include <networkit/distance/DynDijkstra.hpp>
 #include <networkit/distance/DynBFS.hpp>
-#include <networkit/auxiliary/Log.hpp>
-#include <networkit/auxiliary/NumericTools.hpp>
-
+#include <networkit/distance/DynDijkstra.hpp>
+#include <networkit/graph/GraphTools.hpp>
 
+// networkit-format
 namespace NetworKit {
 
-DynApproxBetweenness::DynApproxBetweenness(const Graph& G, const double epsilon, const double delta, const bool storePredecessors, const double universalConstant) : Centrality(G, true),
-storePreds(storePredecessors), epsilon(epsilon), delta(delta), universalConstant(universalConstant) {
+DynApproxBetweenness::DynApproxBetweenness(const Graph &G, double epsilon, double delta,
+                                                   bool storePredecessors, double universalConstant)
+    : Centrality(G, true), epsilon(epsilon), delta(delta), storePreds(storePredecessors),
+      universalConstant(universalConstant) {}
+
+inline count DynApproxBetweenness::DFS(node u, count j, Graph &sccDAG,
+                                           std::vector<bool> &marked, std::vector<node> &sorted) {
+    marked[u] = true;
+    sccDAG.forNeighborsOf(u, [&](node v) {
+        if (!marked[v]) {
+            marked[v] = true;
+            j = DFS(v, j, sccDAG, marked, sorted);
+        }
+    });
+    j--;
+    sorted[j] = u;
+    return j;
 }
 
+count DynApproxBetweenness::computeVDdirected() {
+    clock_t begin = clock();
+    StronglyConnectedComponents scc(G);
+    scc.run();
+    clock_t end = clock();
+    INFO("Time to run dyn conn comp: ", end - begin);
+    begin = clock();
+    count sccNum = scc.numberOfComponents();
+    INFO("Number of nodes: ", G.upperNodeIdBound(), ", number of components: ", sccNum);
+    Graph sccDAG(sccNum, true, true);
```

Comment:
Why are we creating a weighted graph? It does not seem that we are using the weights.